src/include/miopen/median/problem_description.hpp

```diff
@@ -0,0 +1,250 @@
+                             " dimensions");
+        }
+
+        return true;
+    }
+
+    bool IsRightLength() const
+    {
+        auto input_dims   = inputDesc.GetLengths();
+        auto output_dims  = outputDesc.GetLengths();
+        auto indices_dims = indicesDesc.GetLengths();
+
+        auto desired_dims = input_dims;
+        if(keepdim)
+        {
+            desired_dims[dim] = 1;
+        }
+        else
+        {
+            desired_dims.erase(desired_dims.begin() + dim);
+            if(desired_dims.size() == 0)
+                desired_dims.push_back(1);
+        }
+
+        if(output_dims != desired_dims || indices_dims != desired_dims)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Input and output/indices tensor dimension lengths do "
+                         "not match.");
+        }
+
+        return true;
+    }
+
+    bool IsSameType() const
+    {
+        if(inputDesc.GetType() != outputDesc.GetType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Input and output tensor data types do not match.");
+        }
+        return true;
+    }
+
+    bool IsValidFloat() const
+    {
+        return inputDesc.GetType() == miopenFloat || inputDesc.GetType() == miopenHalf ||
+               inputDesc.GetType() == miopenBFloat16;
+    }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+private:
+    const TensorDescriptor& inputDesc;
+    const TensorDescriptor& outputDesc;
+    const TensorDescriptor& indicesDesc;
+    uint64_t dim;
+    bool keepdim;
+};
+
+struct BwdProblemDescription : ProblemDescriptionBase
+{
+    BwdProblemDescription(const TensorDescriptor& outputGradDesc_,
+                          const TensorDescriptor& indicesDesc_,
+                          const TensorDescriptor& inputGradDesc_,
+                          const uint64_t dim_,
+                          const bool keepdim_)
+        : outputGradDesc(outputGradDesc_),
+          indicesDesc(indicesDesc_),
+          inputGradDesc(inputGradDesc_),
+          dim(dim_),
+          keepdim(keepdim_)
+    {
+        IsValidNumDims();
+        IsValidDim();
+        IsRightLength();
+        IsSameType();
+    }
+
+    const TensorDescriptor& GetOutputGradDesc() const { return outputGradDesc; }
+    const TensorDescriptor& GetIndicesDesc() const { return indicesDesc; }
+    const TensorDescriptor& GetInputGradDesc() const { return inputGradDesc; }
+    uint64_t GetDim() const { return dim; }
+    bool GetKeepDim() const { return keepdim; }
+
+    bool IsValidNumDims() const
+    {
+        if(inputGradDesc.GetNumDims() > 5)
+        {
+            MIOPEN_THROW(
+                miopenStatusBadParm,
+                "MedianForward: A tensor with dimensions greater than 5 is not supported yet.");
+        }
+
+        return true;
+    }
```

Comment:
The same as above. Move to `IsApplicable`, do not throw exception.

Reply:
solved